connect/clients/ipfs.py

```diff
@@ -26,61 +27,58 @@ def __init__(self, uri, replication_factor):
         self._get_peers_uri = self._uri + "/peers"
         self._unpin_cid_uri = self._uri + "/pins/"
 
-    async def persist_json_to_ipfs(self, payload_dict):
+    async def persist_json_to_ipfs(self, payload: dict):
         """
         Allows persistence of JSON payloads (as a dict) into an IPFS cluster
         The IPFS cluster instance being connected to is configurable.
 
         Content is pinned by default during the add operation.
 
-        :param payload_dict
+        :param payload: dict containing the message to be persisted to IPFS
         :returns: Tuple containing an HTTP response code and an unique ID
                   addressing the content added to the IPFS cluster. Returns
                   response code and None if a non 200 response code is returned
                   from the IPFS Cluster.
         """
-        response = None
-
-        if not payload_dict:
+        if not payload:
             error_msg = "dict payload not provided for persistence into IPFS"
             logger.error(error_msg)
             raise ValueError(error_msg)
 
         async with httpx.AsyncClient() as client:
             try:
-                with NamedTemporaryFile() as payload_temp_file:
-                    json.dump(payload_dict, payload_temp_file)
+                with NamedTemporaryFile(mode="w+") as payload_temp_file:
+                    json.dump(payload, payload_temp_file, cls=ConnectEncoder)
                     payload_temp_file.flush()
                     temp_file_name = payload_temp_file.name
 
                     files = {
                         "upload-file": (
                             temp_file_name,
                             open(temp_file_name, "r"),
-                            "multipart/form-data",
+                            "application/json",
                         )
                     }
                     response = await client.post(self._add_uri, files=files)
                     response_code = response.status_code
 
                 if response_code == 200:
-                    response_dict = json.loads(response.json())
-                    info_msg = (
-                        "Successfully added and pinned to IPFS -- "
-                        + f"cid: {response_dict['cid']}"
+                    result = response.text.split("\n")
```

Comment:
Is this a response from the httpx.AsyncClient? If so can we use the `json()` method here to streamline things a bit?
https://www.python-httpx.org/quickstart/#json-response-content

Reply:
No, it was using .json(), but that actually does not work because it returns multiple sets of json responses.

Reply:
Ah - got it! Thank you for the clarification!